[DO NOT MERGE] Dev

Roma7-7-7 opened the pull request.

shrralis committed:
- Added fully working Vue.js project.

shrralis committed:
- All of the requests must be passed now.

shrralis committed:
Merge pull request #3 from nromanen/vuejs_impl

Vuejs impl

shrralis committed:
- Fixing Vue.js.

shrralis committed:
- Fixing Vue.js.

shrralis committed:
- Fixing Vue.js.
 - README.md fixes.

shrralis committed:
- Fixing Vue.js.

shrralis committed:
- Fixing Vue.js.

shrralis committed:
- Fixing Vue.js.

shrralis committed:
- Removed unnecessary comments for turning formatter off.

shrralis committed:
- Fixed JavaScript part of `HelloWorld.vue`.

shrralis committed:
Merge pull request #4 from nromanen/vuejs_impl

Vuejs impl

shrralis committed:
Front-end:
 - Added login page.

Back-end:
[none]

shrralis committed:
Front-end:
 - Added registration page.
 - Merged login and registration page into one `auth` page.

Back-end:
[none]

shrralis committed:
Front-end:
 - Finished registration.
 - Added sign in page.
Back-end:
 - Some code.

Commit:
add issue view, issueController

shrralis committed:
Front-end:
 - Finished login page.
 - Now we have User cookie after successfully auth.
 - True CORS implemented.

Back-end:
 - True CORS implemented.
 - Now we can get current user from Security.

Database:
 - Now `map_markers.lat` and `map_markers.lng` is `DOUBLE PRECISION` (`FLOAT8`) instead of `REAL`.

shrralis committed:
Merge pull request #5 from nromanen/vuejs_impl

Vue.js implemented

Commit:
some work with front-end

Commit:
some work with front-end

Commit:
Merge branch 'dev' into view_issue

# Conflicts:
#	front-end/package-lock.json
#	front-end/package.json
#	front-end/src/App.vue
#	front-end/src/router/index.js
#	pom.xml
#	src/main/java/com/shrralis/ssdemo1/configuration/AppConfig.java

Commit:
fix errors after merge

shrralis committed:
# Shrralis commit changelog #33:

    [GLOBAL]:
     - Project has name as "Geo Citizen" now.

    [Front-end]:
     - Added some constants for JsonErrors.
     - Removed `window.href` changing on switching tabs at `AuthPage`.
     - Added "index" page with login/logout links.
     - User doesn't store in cookies anymore, now it uses local storage.
     - Added some automatic getting JsonError message at `SignUpPage`.
     - Added possibility to "guarding" links (components/pages) with `router`.
     - Now we have API url as constant (main.js->`Vue.http.options.root`).

    [Back-end]:
     - DatabaseConfig now scans packages too (with annotation).
     - Added logout success handler at SecurityConfig.
     - @RequestMappings "/login" and "/logout" now has the same method.
     Also that method has been renamed and improved when user isn't logged in.
     - Now we can get user information. Maybe it has some sense to secure that.
     - User entity now is "proxied" as non-lazy (with annotation).
     - EntityNotUniqueException now has `additionalInfo` field.
     (for registration detailing)
     - Fixed NPE in AuthorizedUser.`getCurrent()`.

    [Database]:
     `null`

Commit:
map_impl

Commit:
add minus

Commit:
merge map_impl

Commit:
change port 8181 -> 8080

shrralis committed:
# Shrralis commit changelog #34:

    [GLOBAL]:
     `null`

    [Front-end]:
     - Added Home button to auth forms.

    [Back-end]:
     `null`

    [Database]:
     `null`

Commit:
Merge pull request #6 from nromanen/view_issue

View issue

shrralis committed:
# Shrralis commit changelog #34:

    [GLOBAL]:
     `null`

    [Front-end]:
     - Fixed wrong warnings.

    [Back-end]:
     `null`

    [Database]:
     `null`

shrralis committed:
Merge branch 'dev' into vuejs_impl

# Conflicts:
#	front-end/src/App.vue
#	front-end/src/router/index.js
#	src/test/java/com/shrralis/ssdemo1/repository/UsersRepositoryTest.java

shrralis committed:
# Shrralis commit changelog #34:

    [GLOBAL]:
     `null`

    [Front-end]:
     - Removed unnecessary wrong dependency.

    [Back-end]:
     `null`

    [Database]:
     `null`

shrralis committed:
Merge pull request #7 from nromanen/vuejs_impl

Vuejs impl

shrralis committed:
# Shrralis commit changelog #34:

    [GLOBAL]:
     `null`

    [Front-end]:
     `null`

    [Back-end]:
     `null`

    [Database]:
     `null`

Commit:
map updating bug fixed

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/IssueRestController.java:
```diff
@@ -0,0 +1,65 @@
+@CrossOrigin
+public class IssueRestController {
+    @Autowired
+    IIssueService issueService;
```
Constructor initialization is best practice proposed by Spring framework

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/IssueRestController.java:
```diff
@@ -0,0 +1,65 @@
+        if (vote != null) {
+            return vote;
+        }
+        else return new ResponseEntity<Void>(HttpStatus.NO_CONTENT);
```
I would recommend to create NoContentException, common handler for it and throw it here

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/IssueRestController.java:
```diff
@@ -0,0 +1,65 @@
+
+    @RequestMapping(value = "calculateVote/{issueId}")
+    public Map<String, Integer> calculateVote(@PathVariable("issueId") Integer issueId) {
+        Map<String, Integer> map = new HashMap<>();
```
Use DTO

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/MapRestController.java:
```diff
@@ -0,0 +1,67 @@
+public class MapRestController {
+
+    @Autowired
+	IMapMarkersService service;
```
What's wrong with tabs ?

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/UsersRestController.java:
```diff
@@ -12,33 +12,29 @@
-    }
+	@Secured({ "ROLE_USER", "ROLE_ADMIN" })
+	@RequestMapping("get")
+	public JsonResponse getUserInfo(@RequestParam("id") Integer id) {
```
Make id path variable and use int

shrralis replied:
Done

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/system/ExceptionHandlerControllerAdvice.java:
```diff
@@ -12,38 +12,79 @@
+    public JsonResponse illegalArgumentExceptionHandler(IllegalArgumentException e) {
+	    logger.error("IllegalArgumentException: {}", e.getMessage());
+
+	    Matcher matcher = Pattern.compile("\'([^\']*)\'").matcher(e.getMessage());
```
Do you throw IllegalArgumentException? It might be better to use custom class with message field so we  do not need to parse exception

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/system/ExceptionHandlerControllerAdvice.java:
```diff
@@ -12,38 +12,79 @@
+		logger.error("MissingServletRequestParameterException: {}", e.getMessage());
+
+		if (e.getMessage() != null) {
+			Matcher matcher = Pattern.compile("\'([^\']*)\'").matcher(e.getMessage());
```
MissingServletRequestParameterException has fields that can be used to generate message instead of parsing

shrralis replied:
Thanks

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/system/GeneralRestController.java:
```diff
@@ -0,0 +1,18 @@
+public class GeneralRestController {
+	@RequestMapping("accessDenied")
+	public JsonResponse accessDenied() {
+		return new JsonResponse(JsonError.Error.ACCESS_DENIED);
```
Do you have separate endpoint for accessDenied? I would expect that it will be some exception handler

shrralis replied:
O.K.

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/dto/RegisterUserDTO.java:
```diff
@@ -25,39 +28,39 @@
 
     public String getLogin() {
-        return login;
+	    return login.trim();
```
Use StringUtils.trim . It will prevent from NPE

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/exception/interfaces/AbstractShrralisException.java:
```diff
@@ -0,0 +1,33 @@
+ * @author shrralis (https://t.me/Shrralis)
+ * @version 1.0 Created 12/24/17 at 2:19 AM
+ */
+public abstract class AbstractShrralisException extends Exception {
```
What does Shrralis mean?

shrralis replied:
I had no idea how to call that. Now we decided to rename it as `Abstract Citizen Exception`. Waiting for that with coming commits.

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/AuthProvider.java:
```diff
@@ -50,29 +52,29 @@ public void afterPropertiesSet() {
-            user = userDetailsService.loadUserByUsername(username);
+            userDetails = userDetailsService.loadUserByUsername(login);
         } catch (UsernameNotFoundException e) {
             throw e;
```
Why do we catch this exception i we don't handle it ?

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/AuthProvider.java:
```diff
@@ -85,11 +87,11 @@ private UserDetails retrieveUser(String username) throws AuthenticationException
             String password = authentication.getCredentials().toString();
 
-            if (password.length() > 32 || !passwordEncoder.matches(password, userDetails.getPassword())) {
+            if (password.length() < 8 || !passwordEncoder.matches(password, userDetails.getPassword())) {
```
What is this magic number ?

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/AuthSuccessHandler.java:
```diff
@@ -33,7 +36,11 @@
+	private RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
+	@Autowired
+	private AuthRestController authRestController;
```
Wow wow wow. Autowire controller ? 0_0

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/AuthorizedUser.java:
```diff
@@ -68,7 +72,15 @@ public String toString() {
     }
+
+	public static AuthorizedUser getCurrent() {
+		if (SecurityContextHolder.getContext().getAuthentication() == null) {
```
As I remember it will never return null. At least Anonymous Authentication

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/LogoutSuccessHandler.java:
```diff
@@ -0,0 +1,34 @@
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse,
+			Authentication authentication) throws IOException, ServletException {
+		new ObjectMapper().writeValue(httpServletResponse.getWriter(), authRestController.checkCurrSession(null));
```
It is best practice to make ObjectMapper static final constant

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/service/AuthServiceImpl.java:
```diff
@@ -30,22 +36,63 @@
-                .setSurname(user.getSurname())
-                .build()));
+    public JsonResponse signUp(RegisterUserDTO user) throws AbstractShrralisException {
+	    if (user == null) {
```
Objects.notNull

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/service/AuthServiceImpl.java:
```diff
@@ -30,22 +36,63 @@
+		    throw new NullPointerException("Passed `User` is null");
+	    }
+
+	    if (repository.getByLogin(user.getLogin()) != null) {
```
I would recommend create and use method like isExistWithLogin/Email . There are no need to extract whole entity

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/service/AuthServiceImpl.java:
```diff
@@ -30,22 +36,63 @@
+	    return new JsonResponse(JsonResponse.OK);
     }
+
+	private boolean isLoginValid(String login) {
```
Spring validation

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/service/interfaces/IAuthService.java:
```diff
@@ -13,13 +13,27 @@
  */
 public interface IAuthService {
-    JsonResponse signUp(RegisterUserDTO user);
+	/**
```
It is good practice but don't spend time for javadocs

Roma7-7-7 commented on src/main/java/com/shrralis/tools/TextUtils.java:
```diff
@@ -0,0 +1,102 @@
+	 * @return true OR false
+	 */
+	public static boolean containsIgnoreCase(String str, String searchStr) {
+		if (str == null || searchStr == null) {
```
StringUtild idBlank/isEmpty

Roma7-7-7 commented on src/main/java/com/shrralis/tools/TextUtils.java:
```diff
@@ -0,0 +1,102 @@
+	 *
+	 * @return true OR false
+	 */
+	public static boolean containsIgnoreCase(String str, String searchStr) {
```
https://commons.apache.org/proper/commons-lang/javadocs/api-3.6/org/apache/commons/lang3/StringUtils.html#containsIgnoreCase-java.lang.CharSequence-java.lang.CharSequence-

Roma7-7-7 commented on src/main/java/com/shrralis/tools/TextUtils.java:
```diff
@@ -0,0 +1,102 @@
+	 *
+	 * @return true OR false
+	 */
+	public static boolean isEmpty(String s) {
```
StringUtils

Roma7-7-7 commented on src/main/java/com/shrralis/tools/TextUtils.java:
```diff
@@ -0,0 +1,102 @@
+	 *
+	 * @return true OR false
+	 */
+	public static boolean isEmailValid(String s) {
```
Spring validation, common libs

Roma7-7-7 commented on src/main/java/com/shrralis/tools/TextUtils.java:
```diff
@@ -0,0 +1,102 @@
+	 *
+	 * @return true OR false
+	 */
+	public static boolean isEmptyTrimmed(String s) {
```
StringUtils

Commit:
Entity Issue updated

Commit:
Entity Issue updated

Commit:
Entity Issue updated

Commit:
map updating bug fixed

Commit:
Merge pull request #9 from nromanen/map_impl

Issue entity updated, map bug fixed

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/MapRestController.java:
```diff
@@ -12,56 +12,52 @@
-@CrossOrigin
 @RestController
-@RequestMapping("map")
+@RequestMapping(value = "/map", consumes = MediaType.APPLICATION_JSON_VALUE)
```
Is it possible to set json globally so we will not specify it on every controller ?

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/controller/MapRestController.java:
```diff
@@ -12,56 +12,52 @@
     @Autowired
-	IMapMarkersService service;
+	public MapRestController(IMapMarkersService markerService, IIssueService issueService) {
+    	this.markerService = markerService;
```
Code convention. Missing tab

shrralis committed:
# Shrralis commit changelog #35:

    [GLOBAL]:
     - We're using Java8 now and Spring 4 instead of Java9 and Spring 5.
     - Password recovery added. [need to finish and test with front-end]

    [Front-end]:
     - Added password recovery form.
     - Some code cleans.
     - Added showing some errors (if we retrieved them) from back-end.

    [Back-end]:
     - We have mappers for converting entities into DTO and back.
     - We can send E-Mails now.
     - Added `@Transactional` to all of the services.
     - Now repositories are not as `@Resource`, they're `@Autowired`.
     - Spring Security configuration rearranged a little.
     - Removed unnecessary `GeneralRestController`.
     - Now we have Apache Commons dependency.

    [Database]:
     - New table for password recovery.

shrralis committed:
Merge branch 'dev' into dto_impl

# Conflicts:
#	front-end/src/router/index.js
#	src/main/java/com/shrralis/ssdemo1/controller/IssueRestController.java
#	src/main/java/com/shrralis/ssdemo1/entity/Issue.java
#	src/main/java/com/shrralis/ssdemo1/service/MapMarkersServiceImpl.java
#	src/main/java/com/shrralis/ssdemo1/service/interfaces/IIssueService.java
#	src/main/java/com/shrralis/ssdemo1/service/interfaces/IMapMarkersService.java

shrralis committed:
# Shrralis commit changelog #36:

    [GLOBAL]:
     Merged with last `dev` branch.

    [Front-end]:
     `null`

    [Back-end]:
     `null`

    [Database]:
	 `null`

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     - Recoded with Roma's suggestions. (much thanks)

##    [Front-end]:
     `null`

##    [Back-end]:
     - Refactored some code.
     - New `AuthenticationFailureHandler`.

##    [Database]:
	 - Removed some unnecessary start data from change logs.

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     `null`

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 - Fixed `images.src` length to 128.

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     - Last Roma's suggestions applied.

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 `null`

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     - Last Roma's suggestions applied.

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 `null`

shrralis committed:
Merge pull request #10 from nromanen/dto_impl

Dto impl

shrralis committed:
# Shrralis commit changelog #37:

##    [GLOBAL]:
     - Added some new security logic.

##    [Front-end]:
     - Fixed bug with logout.
     - Added latest error constants.
     - New `md-chip` styles.

##    [Back-end]:
     - Fixed bug with logout.
     - Added new error constants.
     - New checking on failed authentication attempts.
     - New checking on "ddos" with creating recovery tokens.

##    [Database]:
	 - New column in table `users`, that is `failed_auth_count`.

shrralis committed:
# Shrralis commit changelog #37:

##    [GLOBAL]:
     - Added some new security logic.

##    [Front-end]:
     - Fixed bug with logout.
     - Added latest error constants.
     - New `md-chip` styles.

##    [Back-end]:
     - Fixed bug with logout.
     - Added new error constants.
     - New checking on failed authentication attempts.
     - New checking on "ddos" with creating recovery tokens.

##    [Database]:
	 - New column in table `users`, that is `failed_auth_count`.

shrralis committed:
# Shrralis commit changelog #37:

##    [GLOBAL]:
     - Added new README.

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 `null`

shrralis committed:
# Shrralis commit changelog #38:

##    [GLOBAL]:
     `null`

##    [Front-end]:
     - Fixed URL for getting current session user info.

##    [Back-end]:
     - Added new RestController mapping for getting current session user info.

##    [Database]:
	 `null`

shrralis committed:
# Shrralis commit changelog #38:

##    [GLOBAL]:
     - Applied Roma's suggestions.

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 `null`

shrralis committed:
Merge pull request #11 from nromanen/authentication_polish

Authentication polish
